code/game/objects/items/mixing_bowl.dm

```diff
@@ -0,0 +1,159 @@
+
+/obj/item/mixing_bowl
+	name = "mixing bowl"
+	desc = "Mixing it up in the kitchen."
+	flags = OPENCONTAINER
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "mixing_bowl"
+	var/max_n_of_items = 25
+	var/dirty = FALSE
+	var/clean_icon = "mixing_bowl"
+	var/dirty_icon = "mixing_bowl_dirty"
+
+/obj/item/mixing_bowl/New()
+	..()
+	create_reagents(100)
+
+/obj/item/mixing_bowl/attackby(obj/item/I, mob/user, params)
+	if(dirty)
+		if(istype(I, /obj/item/soap))
+			user.visible_message("<span class='notice'>[user] starts to scrub [src].</span>", "<span class='notice'>You start to scrub [src].</span>")
+			if(do_after(user, 20 * I.toolspeed, target = src))
+				clean()
+				user.visible_message("<span class='notice'>[user] has scrubbed [src] clean.</span>", "<span class='notice'>You have scrubbed [src] clean.</span>")
+			return 1
+		else
+			to_chat(user, "<span class='warning'>You should clean [src] before you use it for food prep.</span>")
+		return 0
+	if(is_type_in_list(I, GLOB.cooking_ingredients[RECIPE_MICROWAVE]) || is_type_in_list(I, GLOB.cooking_ingredients[RECIPE_GRILL]) || is_type_in_list(I, GLOB.cooking_ingredients[RECIPE_OVEN]) || is_type_in_list(I, GLOB.cooking_ingredients[RECIPE_CANDY]))
+		if(contents.len>=max_n_of_items)
+			to_chat(user, "<span class='alert'>This [src] is full of ingredients, you cannot put more.</span>")
+			return 1
+		if(istype(I, /obj/item/stack) && I:amount > 1)
+			new I.type (src)
+			I:use(1)
```

Comment:
really.

Reply:
Copied from old code and slipped past me, will refactor later